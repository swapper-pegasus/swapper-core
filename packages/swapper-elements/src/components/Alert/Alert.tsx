import React from 'react'
import { Icon } from '../Icon'

type AlertType = 'warning' | 'error' | 'success';

type Props = {
  title?: string,
  description: string,
  type: AlertType,
  dataTestId?: string,
};

const iconsMapp = {
  warning: 'info',
  error: 'error',
  success: 'success'
}

export function Alert ({
  title,
  description,
  type,
  dataTestId
}: Props) {
  return (
    <div data-testid={dataTestId} className={`flex items-center bg-${type} border-l-4 border-${type}-darkest text-${type}-dark py-2 min-w-full`}>
      <div className="px-2">
        <Icon type={iconsMapp[type]} className='w-6 h-6'/>
      </div>
      <div>
        <p className='font-semibold'>{title}</p>
        <p>{description}</p>
      </div>
    </div>
  )
}
