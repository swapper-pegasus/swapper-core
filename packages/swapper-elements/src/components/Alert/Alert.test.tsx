import React from 'react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import { Alert } from './Alert'

describe('Alert', () => {
  it('warning', () => {
    const tree = renderer.create(<Alert type='warning' title='My title' description='My description'/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('success', () => {
    const tree = renderer.create(<Alert type='success' title='My title' description='My description'/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('error', () => {
    const tree = renderer.create(<Alert type='error' title='My title' description='My description'/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
