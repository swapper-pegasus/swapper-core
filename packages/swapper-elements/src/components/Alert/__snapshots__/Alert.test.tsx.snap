// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert error 1`] = `
<div
  className="flex items-center bg-error border-l-4 border-error-darkest text-error-dark py-2 min-w-full"
>
  <div
    className="px-2"
  >
    <svg
      aria-hidden="true"
      className="w-6 h-6"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
      />
    </svg>
  </div>
  <div>
    <p
      className="font-semibold"
    >
      My title
    </p>
    <p>
      My description
    </p>
  </div>
</div>
`;

exports[`Alert success 1`] = `
<div
  className="flex items-center bg-success border-l-4 border-success-darkest text-success-dark py-2 min-w-full"
>
  <div
    className="px-2"
  >
    <svg
      aria-hidden="true"
      className="w-6 h-6"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
      />
    </svg>
  </div>
  <div>
    <p
      className="font-semibold"
    >
      My title
    </p>
    <p>
      My description
    </p>
  </div>
</div>
`;

exports[`Alert warning 1`] = `
<div
  className="flex items-center bg-warning border-l-4 border-warning-darkest text-warning-dark py-2 min-w-full"
>
  <div
    className="px-2"
  >
    <svg
      aria-hidden="true"
      className="w-6 h-6"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
      />
    </svg>
  </div>
  <div>
    <p
      className="font-semibold"
    >
      My title
    </p>
    <p>
      My description
    </p>
  </div>
</div>
`;
